### PR TITLE
libvirt_pcicontr: Introduce reset_pci_num api

### PR DESCRIPTION
**Pull request by liuyd96**
So cases failed due to error "No more available PCI slots".
As a matter of convenience, Introduce an API to reset the number of
guest's PCI directly.

BTW. Different platforms may require additional configuration, I just
tested it on aarch64 guest.

Signed-off-by: Liu Yiding <liuyd.fnst@fujitsu.com>

### CODE REVIEW

**Comment by liuyd96** [comment]
Hi, @chloerh. Help review it. Thanks.

**Comment by chloerh** [comment]
@smitterl @dzhengfy @kylazhang @chunfuwen This pr is about adding pci slots, you might be interested.

File: virttest/utils_test/libvirt.py [hunk 1]
```diff
@@ -1532,6 +1532,35 @@ def create_controller_xml(contr_dict):
     return contr
 
 
+def reset_pci_num(vm_name, num=10):
```
Comment (chunfuwen) [inline virttest/utils_test/libvirt.py:1535]: suggest change default num= 15, since we see "no more pci slot " issue even number is 10

**Comment by liuyd96** [comment]
Hi, @chloerh @chloerh.
I updated this PR, help review it. Thanks~

File: virttest/utils_libvirt/libvirt_pcicontr.py [hunk 2]
```diff
@@ -51,3 +54,30 @@ def get_free_pci_slot(vm_xml, max_slot=31):
...
+    """
+    vmxml = vm_xml.VMXML.new_from_dumpxml(vm_name)
+    # This func only works on aarch64 and x86/q35 machine
+    if 'aarch64' in vmxml.os.arch \
+            or 'q35' in vmxml.os.machine:
+        # Default pcie setting
+        pcie_root_port = {'controller_model': 'pcie-root-port', 'controller_type': 'pci'}
+        ret_indexes = get_max_contr_indexes(vmxml, 'pci', 'pcie-root-port')
+        cur_pci_num = int(ret_indexes[0])
+        logging.debug("cur_pci_num is %d", cur_pci_num)
```
Comment (dzhengfy) [inline virttest/utils_libvirt/libvirt_pcicontr.py:74]: The logging message could be much formal, like 'The current maximum PCI controller index is: ...'

**Review by dzhengfy — COMMENTED** [review]
Others LGTM

**Review by chloerh — APPROVED** [review]
Agree with @dzhengfy 's suggestion. The rest is ok to me.

File: virttest/utils_libvirt/libvirt_pcicontr.py (hunk 2 above)
Comment (liuyd96) [reply virttest/utils_libvirt/libvirt_pcicontr.py]: Ok, got it. Thanks a lot !

**Review by liuyd96 — COMMENTED** [review]
Hi @dzhengfy @chunfuwen 
I updated this PR, please take a look. Thanks a lot.

**Review by chunfuwen — APPROVED** [review]
lgtm

**Review by dzhengfy — APPROVED** [review]

**Pull request merged**